com.aobuchow.themes.spectrum.preferences/src/com/aobuchow/themes/spectrum/preferences/ColorHSL.java

```diff
@@ -0,0 +1,121 @@
+package com.aobuchow.themes.spectrum.preferences;
+
+import org.eclipse.swt.graphics.Color;
+import org.eclipse.swt.graphics.RGB;
+import org.eclipse.ui.PlatformUI;
+
+public class ColorHSL {
+	private Color color;
+	private float hue;
+	private float saturation;
+	private float luminance;
+
+	public ColorHSL(Color color) {
+		this.color = color;
+		this.hue = color.getRGB().getHSB()[0];
+		this.saturation = color.getRGB().getHSB()[1];
+		this.luminance = color.getRGB().getHSB()[2];
+	}
+
+	public ColorHSL(float hue, float saturation, float luminance) {
+		this(new Color(PlatformUI.getWorkbench().getDisplay(), new RGB(hue, saturation, luminance)));
```

Comment:
I kind of lost track of the Color-dispose-discussion. If Color still needs to be disposed in Eclipse 2020-06, then we would need to make sure that Color objects we create in our bundle are disposed in the end.

Reply:
I assume they have to be disposed in 2020-06 as the API that does not require a display is only available in 2020-09. I’ll try to verify this, but you’re right: the colours need to be disposed.